libs/category/src/drivers/magento/category.service.ts

```diff
@@ -49,23 +50,32 @@ export class DaffMagentoCategoryService implements DaffCategoryServiceInterface
 	}
 	
 	private getProductsQueryVariables(request: DaffCategoryRequest) {
-		return {
-			filter: this.magentoAppliedFiltersTransformer.transform(request.applied_filters),
+		const queryVariables = {
+			filter: this.magentoAppliedFiltersTransformer.transform(request.id, request.applied_filters),
 			search: null,
 			pageSize: request.page_size,
 			currentPage: request.current_page,
 			sort: this.magentoAppliedSortTransformer.transform(request.applied_sort_option, request.applied_sort_direction)
 		}
+
+		if(!queryVariables.pageSize) delete queryVariables.pageSize;
```

Comment:
Instead of deleting after-the-fact, can we build this along the way for the nullable props?

Reply:
sure. Done